snewpdag/plugins/gen/Uniform.py

```diff
@@ -16,26 +16,27 @@
 from snewpdag.values import TimeSeries, TimeHist
 
 class Uniform (Node):
-  def __init__(self, field, rate, **kwargs):
+  def __init__(self, field, rate, tmin = 0, tmax = 10, **kwargs):
     self.field = field
     self.rate = rate
+    self.tmin = tmin
+    self.tmax = tmax
     self.duration = kwargs.pop('duration', 10.0)
     super().__init__(**kwargs)
 
   def alert(self, data):
     if self.field in data:
       v = data[self.field]
-      dt = self.duration
-      if isinstance(v, TimeHist):
-        if dt <= 0.0 or dt > v.duration:
-          dt = v.duration
-      elif not isinstance(v, TimeSeries):
+      if self.tmax > self.duration:
+        self.tmax = self.duration
+      if self.tmin < 0:
```

Comment:
tmin can be negative, i.e., events can be generated before the reference time of the TimeSeries/TimeHist.